blog/views.py

```diff
@@ -0,0 +1,43 @@
+from django.http import HttpResponseRedirect
+from django.views.generic import ListView, DetailView, FormView
+from blog.models import Post, Commentary
+from django import forms
+
+
+class CommentaryForm(forms.ModelForm):
+    class Meta:
+        model = Commentary
+        fields = ["content"]
+        labels = {"content": "Add Commentary"}
+
+
+class HomeView(ListView):
+    model = Post
+    paginate_by = 5
+    template_name = "blog/index.html"
+    queryset = Post.objects.all().order_by("-created_time")
+    queryset.prefetch_related("commentaries")
+
+
+class PostDetailView(DetailView):
+    model = Post
+    queryset = Post.objects.all().prefetch_related("owner")
+    template_name = "blog/post_detail.html"

```

Comment:
Ensure the template name is corrected to 'blog/post-detail.html' to match the previous feedback.